File: backend/app/schemas/observability.py
from pydantic import BaseModel


class ObservabilityCountItem(BaseModel):
    label: str
    value: int


class ObservabilitySummaryResponse(BaseModel):
    system_health_status: str
    counts: list[ObservabilityCountItem]
    most_common_blocked_fields: list[ObservabilityCountItem]
    most_common_missing_inputs: list[ObservabilityCountItem]
    most_frequent_review_reasons: list[ObservabilityCountItem]

File: frontend/src/services/observability.ts
import { apiGet } from "./api";

export type ObservabilityCountItem = {
  label: string;
  value: number;
};

export type ObservabilityMetricItem = {
  label: string;
  value: string;
};

export type ObservabilitySummaryResponse = {
  system_health_status: string;
  counts: ObservabilityCountItem[];
  pilot_usage_metrics: ObservabilityMetricItem[];
  most_common_blocked_fields: ObservabilityCountItem[];
  most_common_missing_inputs: ObservabilityCountItem[];
  most_frequent_review_reasons: ObservabilityCountItem[];
};

export function getObservabilitySummary() {
  return apiGet<ObservabilitySummaryResponse>("/observability/summary");
}